src/main/java/com/dynatrace/hash4j/file/Imohash1_0_2.java

```diff
@@ -0,0 +1,175 @@
+ * limitations under the License.
+ */
+
+/*
+ * This file includes a Java port of the Imohash algorithm originally published
+ * at https://github.com/kalafut/imohash under the following license:
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Jim Kalafut
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.dynatrace.hash4j.file;
+
+import static com.dynatrace.hash4j.util.Preconditions.checkArgument;
+
+import com.dynatrace.hash4j.hashing.HashStream128;
+import com.dynatrace.hash4j.hashing.HashValue128;
+import com.dynatrace.hash4j.hashing.Hashing;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+class Imohash1_0_2 extends AbstractFileHasher128 {
+
+  static final long DEFAULT_SAMPLE_THRESHOLD = 128 * 1024;
+  static final int DEFAULT_SAMPLE_SIZE = 16 * 1024;
+
+  private static final int BUFFER_SIZE = 1024;
+
+  private final HashStream128 hashStream;
+
+  private final long sampleThreshold;
+
+  private final int sampleSize;
+
+  private final byte[] buffer;
+
+  private Imohash1_0_2(int sampleSize, long sampleThreshold) {
+    checkArgument(sampleSize >= 0);
+    checkArgument(sampleSize * 4L <= sampleThreshold);
+    this.sampleThreshold = sampleThreshold;
+    this.sampleSize = sampleSize;
+    this.hashStream = Hashing.murmur3_128().hashStream();
+    this.buffer = new byte[BUFFER_SIZE];
+  }
+
+  static FileHasher128 create() {
+    return new Imohash1_0_2(DEFAULT_SAMPLE_SIZE, DEFAULT_SAMPLE_THRESHOLD);
+  }
+
+  static FileHasher128 create(int sampleSize, long sampleThreshold) {
+    return new Imohash1_0_2(sampleSize, sampleThreshold);
+  }
+
+  private void processBytes(long numBytes, InputStream stream, HashStream128 hashStream)
+      throws IOException {
+    long numBytesRemaining = numBytes;
+    int bufferPos = 0;
+    while (numBytesRemaining > 0) {
+      int numBytesRequested = buffer.length - bufferPos;
```

Comment:
I feel like the names numBytesRemaining and numBytesRequested are swapped.

Reply:
renamed variable for better clarity